packages/block-library/src/navigation/editor.scss

```diff
@@ -15,9 +15,10 @@ $navigation-item-height: 46px;
 
 // Polish the Appender.
 .wp-block-navigation .block-list-appender {
-	margin: 0;
-	display: flex;
-	align-items: center;
+	display: inline-flex;
+	align-self: center;
+	width: $icon-size;
```

Comment:
Because the container can receive focus, unless we explicitly set the width and height of the container, then it's just a big invisible hit area blocking what you were _probably_ trying to click.